src/mono/wasm/debugger/BrowserDebugProxy/ExpressionEvaluator.cs

```diff
@@ -359,39 +399,40 @@ internal static async Task<JObject> CompileAndRunTheExpression(string expression
                 return value;
             }
 
-            IList<JObject> memberAccessValues = await ResolveMemberAccessExpressions(findVarNMethodCall.memberAccesses, resolver, token);
+            IList<JObject> memberAccessValues = await ResolveMemberAccessExpressions(replacer.memberAccesses, resolver, token);
+
+            IList<JObject> identifierValues = await ResolveIdentifiers(replacer.identifiers, resolver, token);
+
+            syntaxTree = replacer.ReplaceVars(syntaxTree, memberAccessValues, identifierValues, null, null);
 
             // eg. "this.dateTime", "  dateTime.TimeOfDay"
-            if (expressionTree.Kind() == SyntaxKind.SimpleMemberAccessExpression && findVarNMethodCall.memberAccesses.Count == 1)
+            if (expressionTree.Kind() == SyntaxKind.SimpleMemberAccessExpression && replacer.memberAccesses.Count == 1)
             {
                 return memberAccessValues[0];
             }
 
-            IList<JObject> identifierValues = await ResolveIdentifiers(findVarNMethodCall.identifiers, resolver, token);
-
-            syntaxTree = findVarNMethodCall.ReplaceVars(syntaxTree, memberAccessValues, identifierValues, null, null);
-
-            if (findVarNMethodCall.hasMethodCalls)
+            if (replacer.hasMethodCalls)
             {
                 expressionTree = syntaxTree.GetCompilationUnitRoot(token);
 
-                findVarNMethodCall.VisitInternal(expressionTree);
-
-                IList<JObject> methodValues = await ResolveMethodCalls(findVarNMethodCall.methodCall, findVarNMethodCall.memberAccessValues, resolver, token);
+                replacer.VisitInternal(expressionTree);
 
-                syntaxTree = findVarNMethodCall.ReplaceVars(syntaxTree, null, null, methodValues, null);
+                IList<JObject> methodValues = await ResolveMethodCalls(replacer, resolver, token);
+                memberAccessValues = await ResolveMemberAccessExpressions(replacer.memberAccesses, resolver, token);
+                identifierValues = await ResolveIdentifiers(replacer.identifiers, resolver, token);
```

Comment:
This will repeat the work for all previous memberAccesses, and identifiers. And it might not even be needed, like if there was no method call on a primitive.

Instead, you can return `(IList<JObject> methodValues, IList<JObject> newMemberAccessValues, IList<JObject> new IdentifierValues)` from `ResolveMethodCalls`. And then use the `memberAccessValues+newMemberAccessValues`, and identifiers, in the call to`ReplaceVars`.

And that would mean that you can remove `UpdateForMethod*`.